src/openapi/streaming/streaming.ts

```diff
@@ -8,71 +8,10 @@ import StreamingOrphanFinder from './orphan-finder';
 import Connection from './connection/connection';
 import * as connectionConstants from './connection/constants';
 import * as streamingTransports from './connection/transportTypes';
-import type {
-    TransportTypes,
-    ConnectionOptions,
-    ConnectionState,
-} from './connection/types';
+import type * as types from './types';
 import type AuthProvider from '../authProvider';
-import type ParserBase from './parser/parser-base';
-import type { IHubProtocol } from '@microsoft/signalr';
 import type { ITransport } from '../transport/transport-base';
-
-export interface RetryDelayLevel {
-    level: number;
-    delay: number;
-}
-
-export interface StreamingConfigurableOptions {
-    /**
-     * Whether the signal-r streaming connection waits for page load before starting
-     */
-    waitForPageLoad?: boolean;
-    /**
-     * The transports to be used in order by signal-r.
-     */
-    transportTypes?: Array<TransportTypes>;
-    /**
-     * The delay in milliseconds to wait before attempting a new connect after signal-r has disconnected
-     */
-    connectRetryDelay?: number;
-    /**
-     * The levels of delays in milliseconds for specific retry counts. Structure: `[{ level:Number, delay:Number }].`
-     */
-    connectRetryDelayLevels?: RetryDelayLevel[];
-    /**
-     * The map of subscription parsers where key is format name and value is parser constructor.
-     */
-    parsers?: Record<string, new (...args: any) => ParserBase>;
-    /**
-     * The map of subscription parser engines where key is format name and value is an engine implementation.
-     */
-    parserEngines?: Record<string, unknown>;
-    transport?: Array<TransportTypes>;
-    /**
-     *  Message serialization protocol used by signalr core
-     */
-    messageProtocol?: Record<string, any>;
-    messageSerializationProtocol?: IHubProtocol;
-}
-
-const OPENAPI_CONTROL_MESSAGE_PREFIX = '_';
-const OPENAPI_CONTROL_MESSAGE_HEARTBEAT = '_heartbeat';
-const OPENAPI_CONTROL_MESSAGE_RESET_SUBSCRIPTIONS = '_resetsubscriptions';
-const OPENAPI_CONTROL_MESSAGE_RECONNECT = '_reconnect';
-const OPENAPI_CONTROL_MESSAGE_DISCONNECT = '_disconnect';
-
-const DEFAULT_CONNECT_RETRY_DELAY = 1000;
-
-const LOG_AREA = 'Streaming';

```

Comment:
I'm not convinced about pulling LOG_AREA out as a constant. In fact I'd say its only worth a constants file if these are used in multiple files - its generally better to keep constants close to where they are used.
In my opinion what we do in Saxotrader where there is a forced constants file, doesn't really help anything.
But if these are used in multiple files, then so be it.

Reply:
Good point